metashare/api/views.py

```diff
@@ -94,6 +98,39 @@ def get_queryset(self):
 
         return Repository.objects.filter(repo_id__isnull=False, repo_id__in=repo_ids)
 
+    @action(methods=["POST"], detail=True, permission_classes=[AllowAny])
+    def hook(self, request, pk=None):
+        repository = self.model.objects.get(pk=pk)
+        valid_signature = False
+        if repository.hook_secret:
+            valid_signature = validate_gh_hook_signature(
+                hook_secret=repository.hook_secret.encode("utf-8"),
+                signature=request.META.get("HTTP_X_HUB_SIGNATURE", ""),
+                message=request.body,
+            )
+        if not valid_signature:
+            return Response(
+                {"error": "Invalid signature."}, status=status.HTTP_403_FORBIDDEN
+            )
+        user = repository.get_a_matching_user()
+        if not user:
+            return Response(
+                {"error": f"No matching user for repository {pk}."},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
```

Comment:
This `user` logic could live in the `forced` conditional.

Reply:
Changes above have rendered this not an option, alas.